pyiron_base/project/generic.py

```diff
@@ -531,15 +531,21 @@ def iter_jobs(self, path=None, recursive=True, convert_to_object=True, status=No
             recursive (bool): search subprojects [True/False] - True by default
             convert_to_object (bool): load the full GenericJob object (default) or just the HDF5 / JobCore object
             status (str/None): status of the jobs to filter for - ['finished', 'aborted', 'submitted', ...]
+            job_type (str/None): job type to filter for, corresponds to the 'hamilton' column in the job table
 
         Returns:
             yield: Yield of GenericJob or JobCore
         """
-        if status is None:
+        if status is None and job_type is None:
             job_id_lst = self.get_jobs(recursive)["id"]
         else:
             df = self.job_table(recursive=True)
-            job_id_lst = list(df[df["status"] == status]["id"]) if not df.empty else []
+            mask = np.ones_like(df.index, dtype=bool)
+            if status is not None:
+                mask &= df["status"] == status
```

Comment:
This would fail for an empty dataframe

Reply:
Ah, empty dataframes, the bane of my programming existence.  Thanks for catching that.  We should make sure, somewhere that even when no jobs exist in a project at least the column headers are always defined, because then the snippet should work.  That's a separate project though.

Reply:
>  We should make sure, somewhere that even when no jobs exist in a project at least the column headers are always defined, because then the snippet should work. That's a separate project though.

I agree!